server/controllers/userAuth.js

```diff
@@ -254,11 +255,10 @@ const recoveryPassword = (req, resp) => {
 
 const updatePassword = (req, resp) => {
   const { password: updatedPassword, passwordConfirmation } = req.body;
-  const { matches } = validator;
   const { token } = req.params;
 
   const sanitizedToken = sanitize(token);
-  const validationError = !matches(updatedPassword, /^[^\s]{4,32}$/);
+  const validationError = !validatePassword(updatedPassword);
   const passwordsNoMatch =
```

Comment:
you don't need to match passwords if the password is not validated. You can send response right away.